lib/MusicBrainz/Server/Data/WebService.pm

```diff
@@ -202,16 +202,16 @@ sub xml_search
         };
     }
 
-    my $format = ($args->{fmt} // "") eq "json" ? "jsonnew" : "xml";
-
-    my $url_ext = "/ws/2/$resource/?" .
-        "max=$limit&type=$resource&fmt=$format&offset=$offset" .
-        "&query=" . uri_escape_utf8($query) . "&dismax=$dismax";
+    my $format = ($args->{fmt} // "") eq "json" ? "mbjson" : "mbxml";
+    my $defType = $dismax ? 'lucene' : 'dismax';
+    my $url_ext = "/$resource/select?" .
+        "rows=$limit&wt=$format&start=$offset" .
+        "&q=" . uri_escape_utf8($query) . "&defType=$defType";
 
     if (DBDefs->LUCENE_X_ACCEL_REDIRECT) {
-        return { redirect_url => '/internal/search/' . DBDefs->LUCENE_SERVER . $url_ext }
+        return { redirect_url => '/internal/search/' . DBDefs->SOLR_SERVER . $url_ext }
```

Comment:
What about using `SEARCH_SERVER`, `SEARCH_X_ACCEL_REDIRECT`, `SEARCH_ENGINE` and create wrappers so we can switch between search engines easily ?